modules/tests/test/modify-vm-template_test.go

```diff
@@ -0,0 +1,209 @@
+				},
+			}),
+		)
+	})
+	Context("modify template sucess", func() {
+		table.DescribeTable("taskrun succeded and template is updated", func(config *testconfigs.ModifyTemplateTestConfig, checkResults func(*templatev1.Template)) {
+			f.TestSetup(config)
+
+			if template := config.TaskData.Template; template != nil {
+				t, err := f.TemplateClient.Templates(template.Namespace).Create(context.TODO(), template, v1.CreateOptions{})
+				Expect(err).ShouldNot(HaveOccurred())
+				f.ManageTemplates(t)
+			}
+
+			runner.NewTaskRunRunner(f, config.GetTaskRun()).
+				CreateTaskRun().
+				ExpectSuccess().
+				ExpectLogs(config.GetAllExpectedLogs()...).
+				ExpectResults(map[string]string{
+					"name":      config.TaskData.TemplateName,
+					"namespace": config.TaskData.TemplateNamespace,
+				})
+
+			updatedTemplate, err := f.TemplateClient.Templates(string(config.TaskData.TemplateNamespace)).Get(context.TODO(), config.TaskData.TemplateName, v1.GetOptions{})
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(updatedTemplate).ToNot(Equal(nil), "new template should exists")
+			f.ManageTemplates(updatedTemplate)
+
+			checkResults(updatedTemplate)
+		},
+			table.Entry("should update template", &testconfigs.ModifyTemplateTestConfig{
+				TaskRunTestConfig: testconfigs.TaskRunTestConfig{
+					ServiceAccount: ModifyTemplateServiceAccountName,
+				},
+				TaskData: testconfigs.ModifyTemplateTaskData{
+					Template:            testtemplate.NewCirrosServerTinyTemplate().Build(),
+					TemplateName:        testtemplate.CirrosTemplateName,
+					CPUCores:            CPUTopologyNumberStr,
+					CPUSockets:          CPUTopologyNumberStr,
+					CPUThreads:          CPUTopologyNumberStr,
+					Memory:              MemoryValue,
+					TemplateLabels:      mockArray,
+					TemplateAnnotations: mockArray,
+					VMAnnotations:       mockArray,
+					VMLabels:            mockArray,
+				},
+			}, func(t *templatev1.Template) {
+				checkValuesInMap(t.Labels)
+				checkValuesInMap(t.Annotations)
+
+				vms, err := decodeVM(t)
+				Expect(err).ShouldNot(HaveOccurred())
+				for _, vm := range vms {
+					checkValuesInMap(vm.Labels)
+					checkValuesInMap(vm.Annotations)
+
+					Expect(vm.Spec.Template.Spec.Domain.CPU.Sockets).To(Equal(CPUTopologyNumber), "cpu sockets should equal")
+					Expect(vm.Spec.Template.Spec.Domain.CPU.Cores).To(Equal(CPUTopologyNumber), "cpu cores should equal")
+					Expect(vm.Spec.Template.Spec.Domain.CPU.Threads).To(Equal(CPUTopologyNumber), "cpu threads should equal")
+					memory := resource.MustParse(MemoryValue)
+					Expect(vm.Spec.Template.Spec.Domain.Resources.Requests.Memory()).To(Equal(&memory), "memory should equal")
+				}
+			}),
+		)
+	})
+})
+
+func checkValuesInMap(mapToCheck map[string]string) {
+	Expect(mapToCheck["newKey"]).To(Equal("value"))
+	Expect(mapToCheck["test"]).To(Equal("true"))
+}
+
+func decodeVM(template *templatev1.Template) ([]*kubevirtv1.VirtualMachine, error) {
```

Comment:
I feel we should try to move this logic to shared since we use it in multiple places